### ISSUE TITLE
AddModuleScore error

### ISSUE THREAD

**Issue by balthasar0810** [satijalab/seurat#4227]
Hi!

I encounter a strange error using AddModuleScore.
myObj <- AddModuleScore(myObj, features = gene.sets, assay = "RNA",name = names(gene.sets))

gene.sets is a list of vectors of gene names (all contained in the RNA assay). Strange thing is, the code worked before. Now I am getting:
Error in `[[<-`(`*tmp*`, colnames(x = features.scores.use), value = features.scores.use) : 
  [[<- defined for objects of type "S4" only for subclasses of environment

What's that supposed to mean?

I reloaded the whole workspace, same error.

Thanks, best, Chris

**Comment by balthasar0810** [comment]
I was able to reproduce the error. It comes from this line in the code:
object[[colnames(x = features.scores.use)]] <- features.scores.use
in the AddModuleScore code.

Why did it work before and where is the problem now??? What am I not seeing? :-D

**Comment by balthasar0810** [comment]
OK: I was able to solve it by loading an older image (with the same object, at least to my knowledge). I don't know what the problem was. I loaded DESeq2 for the R image with the problem. I am really not sure if this was overwriting sth... If anyone has any suggestions on this, I would really appreciate it as this is super time-consuming and annyoing and I would really like to avoid this in the future....
